src/libos/src/vm/vm_manager.rs

```diff
@@ -424,14 +452,26 @@ impl VMManager {
         // accordingly.
 
         // Determine whether need to do mmap. And when possible, determine the returned address
-        // TODO: should fill zeros even when extending a file-backed mapping?
         let (need_mmap, mut ret_addr) = match (flags, size_type) {
             (MRemapFlags::None, SizeType::Growing) => {
+                let vm_initializer_for_new_range = if let Some((backed_file, offset)) =
+                    writeback_file
+                {
+                    // Update writeback file offset
+                    writeback_file = Some((backed_file.clone(), offset + containing_vma.size()));
```

Comment:
May add more comments to make the logic clear.

Reply:
Rewrite the logic with more match cases. Thank you for your review.